odl/tomo/geometry/detector.py

```diff
@@ -894,6 +895,640 @@ def __str__(self):
+            Parameter value(s) at which to evaluate. A sequence of
+            parameters must have length 2.
+
+        Returns
+        -------
+        deriv : `numpy.ndarray`
+            Array representing the derivative vector(s) at ``param``.
+            If ``param`` is a single parameter, the returned array has
+            shape ``(2,)``, otherwise ``param.shape + (2,)``.
+
+        See Also
+        --------
+        surface
+
+        Examples
+        --------
+        The method works with a single parameter, resulting in a single
+        vector:
+
+        >>> part = odl.uniform_partition([-np.pi / 2, -np.pi / 3],
+        ...                              [ np.pi / 2,  np.pi / 3], [20, 10])
+        >>> det = SphericalDetector(
+        ...     part, axes=[(1, 0, 0), (0, 0, 1)], radius = 2)
+        >>> np.round(det.surface_deriv([0, 0]), 10)
+        array([[ 2., -0.,  0.],
+               [ 0.,  0.,  2.]])
+
+        It is also vectorized, i.e., it can be called with multiple
+        parameters at once (or an n-dimensional array of parameters):
+
+        >>> # 2 pairs of parameters, resulting in 3 vectors for each axis
+        >>> deriv = det.surface_deriv([[0, np.pi / 2], [0, np.pi / 3]])
+        >>> np.round(deriv[0], 10)
+        array([[ 2., -0.,  0.],
+               [ 0.,  0.,  2.]])
+        >>> np.round(deriv[1], 2)
+        array([[ 0.  , -1.  ,  0.  ],
+               [-1.73,  0.  ,  1.  ]])
+        >>> # Pairs of parameters in a (4, 5) array each
+        >>> param = (np.zeros((4, 5)), np.zeros((4, 5)))  # pairs of params
+        >>> det.surface_deriv(param).shape
+        (4, 5, 2, 3)
+        """
+        squeeze_out = (np.broadcast(*param).shape == ())
+        param_in = param
+        param = tuple(np.array(p, dtype=float, copy=False, ndmin=1)
+                      for p in param)
+        if self.check_bounds and not is_inside_bounds(param, self.params):
+            raise ValueError('`param` {} not in the valid range '
+                             '{}'.format(param_in, self.params))
+
+        deriv_phi = np.empty(param[0].shape + (3,))
+        deriv_phi[..., 0] = -np.sin(param[0]) * np.cos(param[1])
+        deriv_phi[..., 1] = -np.cos(param[0]) * np.cos(param[1])
+        deriv_phi[..., 2] = 0
+        deriv_phi *= self.radius
+        deriv_theta = np.empty(param[0].shape + (3,))
+        deriv_theta[..., 0] = -np.cos(param[0]) * np.sin(param[1])
+        deriv_theta[..., 1] = np.sin(param[0]) * np.sin(param[1])
+        deriv_theta[..., 2] = np.cos(param[1])
+        deriv_theta *= self.radius
+        deriv = np.stack((deriv_phi, deriv_theta), axis=-2)
+        deriv = np.matmul(deriv, np.transpose(self.rotation_matrix))
+
+        if squeeze_out:
+            deriv = deriv.squeeze()
+
+        return deriv
+
+    def surface_measure(self, param):
+        """Return the arc length measure at ``param``.
+
+        This is a constant function evaluating to (`radius`, `radius`)
```

Comment:
Should be scalar

Reply:
Is it OK, to remove this function and rely on the default implementation in the Detector class?

Reply:
I am afraid I dont really understand what this function is for, should it measure cell area?

Reply:
Yes, usually the default implementation should be okay. The purpose of this function would be to integrate over the manifold described by the geometry. Not sure whether we really ever need that, maybe it was just too easy to add :roll_eyes: